src/app/CommandSender.h

```diff
@@ -398,8 +415,10 @@ class CommandSender final : public Messaging::ExchangeDelegate
         VerifyOrReturnError(!CommandDataT::MustUseTimedInvoke() || aAddRequestDataParams.timedInvokeTimeoutMs.HasValue(),
                             CHIP_ERROR_INVALID_ARGUMENT);
 
-        return AddRequestDataInternal(aCommandPath, aData, aAddRequestDataParams);
+        DataModel::EncodableType<CommandDataT> encodable(aData);
+        return AddRequest(aCommandPath, encodable, aAddRequestDataParams);
```

Comment:
So why is the new API named `AddRequest`, not `AddRequestData` like it was in the original version of this PR?

Reply:
Perhaps this is code smell, but when `AddRequest` is named `AddRequestData` there is a similarity in parameters that causes issues:

```
    CHIP_ERROR AddRequestData(const CommandPathParams & aCommandPath, const DataModel::EncodableToTLV & aEncodable,
                          AddRequestDataParameters & aAddRequestDataParams);

    template <typename CommandDataT>
    CHIP_ERROR AddRequestData(const CommandPathParams & aCommandPath, const CommandDataT & aData,
                              AddRequestDataParameters & aAddRequestDataParams)
```

Say there is a `SimpleTLVPayload` that implements `app::DataModel::EncodableToTLV`, when you have code like :
```
SimpleTLVPayload simple_tlv_payload;
command_sender.AddResponseData(path, simple_tlv_payload, request_data_params);
```
Code fails to compile because it deduces the templated version of AddResponseData. I thought that having clients implement boiler plate code like this:
```
SimpleTLVPayload simple_tlv_payload;
app::DataModel::EncodableToTLV & encodable(simple_tlv_payload);
command_sender.AddResponseData(path, encodable, request_data_params);
```
was too cumbersome, so I went with rename for the non-templated version of the call

Reply:
Ah, overload resolution ends up not doing the right thing for us here?  OK.

Was this problem there also with the previous PR version?

Reply:
Sounds like it was.  How do you feel about this:

```
    template <typename CommandDataT, typename std::enable_if_t<!std::is_base_of_v<DataModel::EncodableToTLV, CommandDataT>, int> = 0>
    CHIP_ERROR AddRequestData(const CommandPathParams & aCommandPath, const CommandDataT & aData,
                              AddRequestDataParameters & aAddRequestDataParams)
```

and then calling the new thing also AddRequestData?  This should work, I believe (tested locally).

Reply:
Made the change in this PR. It seem to work for me on my local changes thus far.